### PR TITLE
Issue 337

### LINKED ISSUE #337: SequenceProfile does not check input parameters
Currently, the constructor of `SequenceProfile` does not check the shapes of the parameters given to the constructor. This may lead to unexpected errors later: For example, the alphabet is not able to decode a symbol code in `to_consensus()` if the alphabet is shorter than the number of columns in the symbol symbol frequency table.

### PR DESCRIPTION
**Pull request by MaxGreil**
Fixes #337

Implements additional checks for profile constructor.

@padix-key , can you have a look?

### CODE REVIEW

File: src/biotite/sequence/profile.py [hunk 1]
```diff
@@ -87,6 +87,20 @@ def __init__(self, symbols, gaps, alphabet):
         self._gaps = gaps
         self._alphabet = alphabet
 
+        if len(alphabet) != self.symbols.shape[1]:
+            raise ValueError(
+                f"The given alphabet is shorter ({len(self.alphabet)}) "
```
Comment (padix-key) [inline src/biotite/sequence/profile.py:92]: It could also be the case, that the alphabet is longer than the number of columns.

**Review by padix-key — CHANGES_REQUESTED** [review]
Thanks @MaxGreil . I think there is a potentially misleading formulation in the error message.

**Comment by MaxGreil** [comment]
Hi @padix-key , I revised the error text. Can you have another look?

**Review by padix-key — APPROVED** [review]
Looks good to me, thank you!

**Pull request merged**